### PR TITLE
feat: [CDS-40818]: PowerShell support for ShellScript and CustomApproval steps

### PR DESCRIPTION
**Pull request by filip5rovic**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36118)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Vulnerability 1 Vulnerabilities 
 * Code issue 6 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36118)

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/WinRmShellScriptTaskNG.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+import java.time.Instant;
+import java.util.Collections;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.NotImplementedException;
+
+@Slf4j
+public class WinRmShellScriptTaskNG extends AbstractDelegateRunnableTask {
+  public static final String COMMAND_UNIT = "Execute";
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/WinRmShellScriptTaskNG.java:47]: Please reuse this from ShellScript itself.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by filip5rovic** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/WinRmShellScriptTaskNG.java (hunk 1 above)
Comment (filip5rovic) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/shell/WinRmShellScriptTaskNG.java]: Done 👍

**Comment by filip5rovic** [comment]
trigger feature-build

**Comment by filip5rovic** [comment]
@yogesh-chauhan 
Answers for https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist:
1. Yes
2. N/A
3. Yes, Yes, Yes
4. No changes in existing tasks
5. No enums removed
6. Yes, task type, added at the end

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by yogesh-chauhan** [comment]
trigger codebasehash

**Comment by filip5rovic** [comment]
trigger publish-delegate

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 10 of 18 files at r1, 3 of 6 files at r2, 1 of 3 files at r3, 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36118)* status: :shipit: complete! ✓

**Comment by puthrayaharness** [comment]
trigger codebasehash

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 8 of 18 files at r1, 1 of 6 files at r2, 2 of 6 files at r4, 15 of 15 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36118)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]

**Review by poojaSinghal — COMMENTED** [review]
> Reviewed 3 of 18 files at r1, 1 of 6 files at r4, 11 of 15 files at r5, 10 of 10 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36118)* status: :shipit: complete! ✓

**Review by poojaSinghal — APPROVED** [review]
> Reviewed 1 of 18 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36118)* status: :shipit: complete! ✓

**Pull request merged**